scripts/git_changes_utils_test.py

```diff
@@ -0,0 +1,567 @@
+        subprocess_swap = self.swap(
+            common, 'start_subprocess_for_result',
+            mock_start_subprocess_for_result)
+
+        with subprocess_swap, self.assertRaisesRegex(
+            ValueError, 'test_oppia_error'
+        ):
+            git_changes_utils.git_diff_name_status(
+                'left', 'right', diff_filter='filter')
+
+    def test_git_diff_name_status_with_no_left_and_right(self) -> None:
+        def mock_start_subprocess_for_result(
+            cmd_tokens: List[str]
+        ) -> Tuple[bytes, None]:
+            if cmd_tokens == ['git', 'diff', '--name-status']:
+                return (b'M\tfile1\nA\tfile2', None)
+            return (b'', None)
+        subprocess_swap = self.swap(
+            common, 'start_subprocess_for_result',
+            mock_start_subprocess_for_result)
+
+        with subprocess_swap:
+            self.assertEqual(
+                git_changes_utils.git_diff_name_status(),
+                [
+                    git_changes_utils.FileDiff(status=b'M', name=b'file1'),
+                    git_changes_utils.FileDiff(status=b'A', name=b'file2')
+                ]
+            )
+
+    def test_git_diff_name_status_with_empty_left_should_error(self) -> None:
+        with self.assertRaisesRegex(
+            ValueError,
+            'Error: left should not be an empty string.'
+        ):
+            git_changes_utils.git_diff_name_status(left='')
+
+    def test_git_diff_name_status_with_empty_right_should_error(self) -> None:
+        with self.assertRaisesRegex(
+            ValueError,
+            'Error: right should not be an empty string.'
+        ):
+            git_changes_utils.git_diff_name_status(right='')
+
+    def test_git_diff_name_status_with_empty_diff_filter_should_error(self) -> None: # pylint: disable=line-too-long
+        with self.assertRaisesRegex(
+            ValueError,
+            'Error: diff_filter should not be an empty string.'
+        ):
+            git_changes_utils.git_diff_name_status(diff_filter='')
+
+    def test_check_file_inside_directory_should_be_false(self) -> None:
+        self.assertFalse(
+            git_changes_utils.check_file_inside_directory(
+                '/usr/directory/test.py', '/usr/opensource/oppia'))
+
+    def test_check_file_inside_directory_should_be_true(self) -> None:
+        self.assertTrue(
+            git_changes_utils.check_file_inside_directory(
+                '/usr/opensource/oppia/test.py', '/usr/opensource/oppia'))
+
+    def test_compare_to_remote(self) -> None:
+        check_function_calls = {
+            'start_subprocess_for_result_is_called': False,
+            'git_diff_name_status_is_called': False,
+            'get_merge_base_is_called': False,
+        }
+        expected_check_function_calls = {
+            'start_subprocess_for_result_is_called': True,
+            'git_diff_name_status_is_called': True,
+            'get_merge_base_is_called': True,
+        }
```

Comment:
Why not just use `swap_with_checks` instead? You can leave the expected args/kwargs params blank if you want to only assert that the mock function gets called

https://github.com/oppia/oppia/blob/654966d11aaf5fe4d69215c6962570b5ef645d70/core/tests/test_utils.py#L1720

ditto elsewhere

Reply:
Done.